Index: app/src/main/java/com/example/appdevelopment/TodoAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/appdevelopment/TodoAdapter.java b/app/src/main/java/com/example/appdevelopment/TodoAdapter.java
--- a/app/src/main/java/com/example/appdevelopment/TodoAdapter.java	(date 1652867298575)
+++ b/app/src/main/java/com/example/appdevelopment/TodoAdapter.java	(date 1652867298575)
@@ -2,82 +2,154 @@
 
 import android.app.Notification;
 import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
 import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
 import android.widget.ImageButton;
+import android.widget.Spinner;
+import android.widget.Switch;
 import android.widget.TextView;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
+import androidx.cardview.widget.CardView;
+import androidx.fragment.app.FragmentActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
-
-import com.firebase.client.core.Context;
+import android.content.Context;
+import com.firebase.ui.database.FirebaseArray;
 import com.firebase.ui.database.FirebaseRecyclerAdapter;
 import com.firebase.ui.database.FirebaseRecyclerOptions;
-import com.google.firebase.installations.Utils;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseException;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class TodoAdapter extends RecyclerView.Adapter<TodoAdapter.todoViewholder> {
 
+    ArrayList<Todo> todoList;
+    Context context;
+    Button saveTODO;
+    Spinner repeatWhen;
+    TextView displayDate, displayTime;
+    DatabaseReference database, userData;
+    RecyclerView fullTodoList;
+    Switch remind;
 
-public class TodoAdapter extends FirebaseRecyclerAdapter<Todo, TodoAdapter.todoViewholder> {
+    public TodoAdapter(Context context, ArrayList<Todo> todoList) {
+        this.context = context;
+        this.todoList = todoList;
+    }
 
-    private static final String TAG = FragmentTODO.class.getSimpleName();
-        public TodoAdapter(@NonNull FirebaseRecyclerOptions<Todo> options) {
-            super(options);
-        }
+    public todoViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.todo_display, parent, false);
+        return new todoViewholder(view);
+    }
 
-        @Override
-        protected void onBindViewHolder(@NonNull todoViewholder holder, int position, @NonNull Todo todo) {
-            holder.todoTitle.setText(todo.getMessage());
-            holder.edit.setOnClickListener(v -> {
-                //onItemClickListener.onItemClick(position);
-            });
-            holder.delete.setOnClickListener(v -> {
-                SimpleItemViewHolder viewHolder = (SimpleItemViewHolder) holder;
-                holder.position = position;
-                Todo todoItem = todoList.get(position);
-                holder.title.setText(todoItem.getName());
-            });
-        }
-
-            public final  class SimpleItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
-                TextView title;
-                public int position;
-                public SimpleItemViewHolder(View itemView) {
-                    super(itemView);
-                    itemView.setOnClickListener(this);
-                    title = (TextView) itemView.findViewById(R.id.myTextView);
-                }
-
-                @Override
-                public void onClick(View view) {
+
+    public void onBindViewHolder(@NonNull todoViewholder holder, int position) {
+        final Todo todo = todoList.get(position);
+        FragmentTODO todoFrag = new FragmentTODO();
+        holder.todoTitle.setText(todoList.get(position).getMessage());
+        //holder.remind
+        holder.todoTitle.setText(todo.getMessage());
+        holder.setItemClickListener(new ItemClickListener(){
+            @Override
+            public void onItemClick(int pos){
+                pos -> todoFrag.editTodo(todo.getAlert(), todo.getAlertDate(), todo.getAlertTime(), todo.getMessage(), todo.getRepeat()));
+            }
+        }
 
-                }
-            }
-        }
+    }
+    private void openDetailActivity(String...details)
+    {
+        Intent i=new Intent(context,getTodoDetail());
+
+        i.putExtra("NAME_KEY",details[0]);
+        i.putExtra("DESC_KEY",details[1]);
+        i.putExtra("PROP_KEY",details[2]);
+
+        getTodoDetail(i);
+    }
+    public int getItemCount(){
+        return todoList.size();
+    }
+    protected void getTodoDetail(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+
+        nameTxt = (TextView) findViewById(R.id.nameDetailTxt);
+        descTxt= (TextView) findViewById(R.id.descDetailTxt);
+        propTxt = (TextView) findViewById(R.id.propellantDetailTxt);
+
+        //GET INTENT
+        Intent i=this.getIntent();
+
+        //RECEIVE DATA
+        String name=i.getExtras().getString("NAME_KEY");
+        String desc=i.getExtras().getString("DESC_KEY");
+        String propellant=i.getExtras().getString("PROP_KEY");
 
-        @NonNull
-        @Override
-        public todoViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.todo_display, parent, false);
-            return new todoViewholder(view);
-        }
+        //BIND DATA
+        nameTxt.setText(name);
+        descTxt.setText(desc);
+        propTxt.setText(propellant);
+
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                        .setAction("Action", null).show();
+            }
+        });
+    }
 
-        static class todoViewholder extends RecyclerView.ViewHolder implements View.OnClickListener{
-            TextView todoTitle;
-            ImageButton delete, edit;
-            public todoViewholder(@NonNull View itemView) {
-                super(itemView);
-                todoTitle = itemView.findViewById(R.id.todoTitle);
-                delete = itemView.findViewById(R.id.btnDelete);
-                edit = itemView.findViewById(R.id.btnEditTodo);
-            }
-            @Override
-            public void onClick(View view) {
+    class todoViewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        TextView todoTitle;
+        ImageButton delete, edit;
+        ItemClickListener itemClickListener;
+
+        public todoViewholder(@NonNull View itemView) {
+            super(itemView);
+            todoTitle = itemView.findViewById(R.id.todoTitle);
+            delete = itemView.findViewById(R.id.btnDelete);
+            edit = itemView.findViewById(R.id.btnEditTodo);
+            repeatWhen = itemView.findViewById(R.id.drpRepeat);
+            remind = itemView.findViewById(R.id.togReminder);
+            displayDate = itemView.findViewById(R.id.txtDisplayDate);
+            displayTime = itemView.findViewById(R.id.txtDisplayTime);
+            saveTODO = itemView.findViewById(R.id.btnSaveToDo);
+            fullTodoList = itemView.findViewById(R.id.todoList);
+
+            itemView.setOnClickListener(this);
+        }
+        public void setItemClickListener (ItemClickListener itemClickListener){
+            this.itemClickListener=itemClickListener;
+        }
+        @Override
+        public void onClick(View view) {
+            this.itemClickListener.onItemClick(this.getLayoutPosition());
+        }
+    }
+    public interface ItemClickListener {
 
-            }
-        }
+        void onItemClick(int pos);
+    }
 }
 
 
